src/components/forms/NetworkDevicesForm/NetworkDeviceIPAddress.tsx

```diff
@@ -0,0 +1,75 @@
+import { type FC } from "react";
+import type { InstanceAndProfileFormikProps } from "../instanceAndProfileFormValues";
+import { Input } from "@canonical/react-components";
+import { type LxdNetwork } from "types/network";
+import { type LxdNicDevice } from "types/device";
+import { typesWithNicStaticIPSupport } from "util/networks";
+
+interface Props {
+  formik: InstanceAndProfileFormikProps;
+  index: number;
+  network?: LxdNetwork;
+  device: LxdNicDevice;
+  readOnly: boolean;
```

Comment:
```suggestion
```
I'd suggest keeping the interface smaller, and inside this we can read it as `const readOnly = formik.values.readonly`.